### PR TITLE
[Telemetry Tables] Fix sort issues

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes https://github.com/nasa/openmct/issues/7809

### Describe your changes:

There was an issue where immutable objects sort order was not being set correctly in telemetry tables. This is because their configuration couldn't be saved and the sort order was not being saved in memory. There was another issue where telemetry was not being re-requested when the sort order of a table was changed and the table was in performance (limited) mode. This would result in incorrect data.

We've moved sort order to both configuration and in-memory and we will re-request telemetry if changing sort order in performance mode.  

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes? #7876 
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/telemetryTable/TelemetryTable.js [hunk 1]
```diff
@@ -131,17 +136,14 @@ export default class TelemetryTable extends EventEmitter {
     this.tableRows = new TableRowCollection();
 
     //Fetch any persisted default sort
-    let sortOptions = this.configuration.getConfiguration().sortOptions;
+    let configSortOptions = this.configuration.getConfiguration().sortOptions;
```
Comment (ozyx) [inline src/plugins/telemetryTable/TelemetryTable.js:139]: let's make it `const`

File: src/plugins/telemetryTable/TelemetryTable.js [hunk 2]
```diff
@@ -173,7 +175,7 @@ export default class TelemetryTable extends EventEmitter {
     this.removeTelemetryCollection(keyString);
 
     let sortOptions = this.configuration.getConfiguration().sortOptions;
-    requestOptions.order = sortOptions?.direction ?? ORDER.DESCENDING; // default to descending
+    requestOptions.order = sortOptions?.direction ?? this.sortOptions.direction; // default to descending
```
Comment (ozyx) [inline src/plugins/telemetryTable/TelemetryTable.js:178]: the comment on this line doens't seem relevant anymore

**Review by ozyx — CHANGES_REQUESTED** [review]
Couple minor suggestions

File: src/plugins/telemetryTable/TelemetryTable.js (hunk 1 above)
Comment (jvigliotta) [reply src/plugins/telemetryTable/TelemetryTable.js:139]: good call! done

File: src/plugins/telemetryTable/TelemetryTable.js (hunk 2 above)
Comment (jvigliotta) [reply src/plugins/telemetryTable/TelemetryTable.js:178]: truth bomb. done!

File: src/plugins/telemetryTable/TelemetryTable.js [hunk 3]
```diff
@@ -131,17 +136,14 @@ export default class TelemetryTable extends EventEmitter {
...
-    let sortOptions = this.configuration.getConfiguration().sortOptions;
+    const configSortOptions = this.configuration.getConfiguration().sortOptions;
 
-    //If no persisted sort order, default to sorting by time system, descending.
-    sortOptions = sortOptions || {
-      key: this.openmct.time.getTimeSystem().key,
-      direction: ORDER.DESCENDING
-    };
+    //If no persisted sort order, use the in-memory sort options
+    this.sortOptions = configSortOptions || this.sortOptions;
```
Comment (ozyx) [inline src/plugins/telemetryTable/TelemetryTable.js:142]: excellent

**Review by ozyx — APPROVED** [review]
ship it

**Review by akhenry — CHANGES_REQUESTED** [review]
Made changes suggested in review and pushed.

File: src/plugins/telemetryTable/TelemetryTable.js [hunk 4]
```diff
@@ -442,11 +444,18 @@ export default class TelemetryTable extends EventEmitter {
   }
 
   sortBy(sortOptions) {
-    this.tableRows.sortBy(sortOptions);
+    this.sortOptions = sortOptions;
```
Comment (akhenry) [inline src/plugins/telemetryTable/TelemetryTable.js:447]: The TelemetryTableConfiguration class exists to try and abstract away some of the boiletplate involved in reconciling in-memory and persisted configuration. Maybe we could move some of this boilerplate there?

eg.
```
getSortOptions() {
    return this.#sortOptions || 
        this.getConfiguration().sortOptions ||
        {
              key: this.openmct.time.getTimeSystem().key,
              direction: ORDER.DESCENDING
        }
}
setSortOptions(sortOptions) {
   this.sortOptions = sortOptions;

   if (this.openmct.editor.isEditing()) {
      let configuration = this.configuration.getConfiguration();
      configuration.sortOptions = sortOptions;
      this.configuration.updateConfiguration(configuration);
    }
}
```

**Review by akhenry — CHANGES_REQUESTED** [review]

**Review by akhenry — APPROVED** [review]

**Pull request merged**